system/Helpers/array_helper.php

```diff
@@ -26,7 +26,11 @@
 	 */
 	function dot_array_search(string $index, array $array)
 	{
-		$segments = explode('.', rtrim(rtrim($index, '* '), '.'));
+		$segments = preg_split('/(?<!\\\)\./', rtrim($index, '* '), 0, PREG_SPLIT_NO_EMPTY);
+
+		$segments = array_map(function ($key) {
```

Comment:
```suggestion
		$segments = array_map(static function ($key) {
```

Reply:
Why did you suggest using ``static``?

Reply:
Using static here is for micro-optimization.